simulator/ns-3.35/src/point-to-point/model/switch-mmu.cc

```diff
@@ -345,9 +381,25 @@ bool SwitchMmu::CheckEgressAdmission(uint32_t port, uint32_t qIndex, uint32_t ps
 }
 
 void SwitchMmu::UpdateIngressAdmission(uint32_t port, uint32_t qIndex, uint32_t psize, uint32_t type) {
-	ingress_bytes[port][qIndex] += psize;
-	ingressPoolUsed += psize;
+	
+	// NOTE: ingress_bytes simple counts total bytes occupied by port, qIndex, 
+	// This includes bytes from ingresspool as well as from headroom. ingress_bytes[port][qIndex] - xoffUsed[port][qIndex] gives us the occupancy in ingressPool.
+	ingress_bytes[port][qIndex] += psize; 
+	totalUsed += psize;
 	totalSharedUsed += psize; // IMPORTANT: totalSharedUsed is only updated in the ingress. No need to update in egress. Avoid double counting.
+
+	/* Update the total headroom used.*/
+	xoffTotalUsed -= xoffUsed[port][qIndex]; // First remove the previously used headroom
```

Comment:
This line is very confusing though I understand what information you want to convey. Can you explicitly calculate delta for headroom buffer explicitly?